apps/eth/views.py

```diff
@@ -106,7 +105,9 @@ def list(self, request, *args, **kwargs):
         log_metric('transmission.info', tags={'method': 'transaction.list', 'module': __name__})
 
         shipment_pk = kwargs.get('shipment_pk', None)
-        if shipment_pk:
+        permission_link = request.query_params.get('permission_link', None)
+
+        if shipment_pk and permission_link or shipment_pk:
```

Comment:
Indeed at first look that statement seems confusing. But it's structured that way to avoid doing multiple `if{}elif{}else` and have in all those leafs the same queryset statement.
Since the transaction permission link is applied on the nested route, the if statement means in case we have both `shipment_pk` and `permission_link` OR we just have `shipment_pk` (meaning the call is for an authenticated user we the due accesses), return the transactions related to that shipment.